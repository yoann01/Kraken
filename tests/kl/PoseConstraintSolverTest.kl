
require Kraken;

operator entry(){
  PoseConstraintSolver solver();

  Boolean debug     = true;
  Scalar rigScale   = 1.0;

  Xfo constrainer();
  constrainer.tr = Vec3(0, 5, 0);
  constrainer.ori.setFromAxisAndAngle(Vec3(0, 1, 0), Math_degToRad(45));

  Mat44 constrainee = Xfo().toMat44();

  solver.solve(debug, rigScale, constrainer.toMat44(), constrainee);

  report(constrainee.translation());
  report(Math_radToDeg(Quat(constrainee.upperLeft()).toEulerAngles().y));
}
