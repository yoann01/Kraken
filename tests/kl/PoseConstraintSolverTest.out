{x:+0.0,y:+5.0,z:+0.0}+45.0